packages/cli/src/cmds/sequenceDiagram/readDiagramFile.ts

```diff
@@ -1,7 +1,20 @@
 import { readFile } from 'fs/promises';
-import { Diagram, unparseDiagram } from '@appland/sequence-diagram';
+import {
+  unparseDiagram,
+  validateDiagram,
+  ValidationResult,
+  Result,
+} from '@appland/sequence-diagram';
 
-export async function readDiagramFile(fileName: string): Promise<Diagram> {
+export async function readDiagramFile(fileName: string): Promise<Result> {
```

Comment:
Thanks for creating the validate diagram function. We don’t want readDiagramFile to always validate first, because in some contexts the input is already known to be a valid diagram and we don’t want to spend the extra computation to validate it. 

So, readDiagramFile and validateDiagram can be two separate functions. 

The sequence diagram CLI can call validate followed by read. In other contexts, read can be called without validate. 

The CLI should have a command option to skip validation, but it should be enabled by default.

Reply:
Addressed.